dace/libraries/blas/nodes/gemm.py

```diff
@@ -425,6 +425,575 @@ def expansion(node, state, sdfg):
+                            If set to None, no tiling is used, corresponding to setting the tile size
+                            equal to the number of columns of B/C.
+        :return:
+        '''
+
+        ((edge_a, outer_array_a, shape_a, strides_a), (edge_b, outer_array_b,
+                                                       shape_b, strides_b),
+         cdata) = _get_matmul_operands(node, parent_state, parent_sdfg)
+
+        dtype_a = outer_array_a.dtype.type
+        dtype_b = outer_array_b.dtype.type
+        dtype_c = dace.DTYPE_TO_TYPECLASS[np.result_type(dtype_a, dtype_b).type]
+        shape_c = (shape_a[0], shape_b[1])
+        if node.transA:
+            raise NotImplementedError(
+                "GEMM FPGA expansion not implemented for transposed A.")
+        if node.transB:
+            raise NotImplementedError(
+                "GEMM FPGA expansion not implemented for transposed B.")
+
+        if outer_array_a.veclen > 1:
+            raise NotImplementedError(
+                "Vectorization not support for input array A.")
+
+        if len(shape_a) != 2 or len(shape_b) != 2 or shape_a[1] != shape_b[0]:
+            raise SyntaxError("Matrix sizes must match")
+
+        ######################################################################
+        # GEMM Parameters and checks
+
+        # Note: the following sizes consider also vectorization
+        vec_width = outer_array_b.dtype.veclen
+        vec_type = dace.vector(dtype_c, vec_width)
+        N, K, M = shape_a[0], shape_a[1], shape_b[1]
+
+        P = num_pes
+        T = tile_size_m
+        if T is None:
+            T = M
+
+        # we will perform sanity check using T and M. But at this stage, we still
+        # don't know to what outer symbol they will map.
+        # We try to resolve them to constant if they are symbolic, otherwise we skip the checks
+        def resolve_symbol_to_constant(symb: dace.symbolic.symbol, parent_sdfg):
+            if not dace.symbolic.issymbolic(symb):
+                return symb
+            else:
+                sdfg = parent_sdfg
+                while sdfg is not None:
+                    if not dace.symbolic.issymbolic(symb, sdfg.constants):
+                        return dace.symbolic.evaluate(symb, sdfg.constants)
+                    else:
+                        sdfg = sdfg.parent_sdfg
+                # can not be resolved
+                return None
+
+        T_constant = resolve_symbol_to_constant(T, parent_sdfg)
+        K_constant = resolve_symbol_to_constant(K, parent_sdfg)
+
+        # Safe delay: this will be used in the compute state, pipeline scope, to insert
+        # a delay between accumulation on the same result if needed.
+        # Further explanations are provided in the compute state.
+
+        # TODO: this is a platform and type dependent parameter. How we can deal with this?
+        if T_constant is not None:
+            L = max(11 - T_constant, 0)
```

Comment:
How about putting 16 as a more conservative number? This should cover even double addition.